src/django_components/component.py

```diff
@@ -897,18 +892,7 @@ def render(self, context: Context) -> str:
         args = safe_resolve_list(context, self.args)
         kwargs = self.kwargs.resolve(context)
 
-        is_default_slot = len(self.fill_nodes) == 1 and self.fill_nodes[0].is_implicit
-        if is_default_slot:
-            slots: Dict[str, SlotFunc] = {
-                DEFAULT_SLOT_KEY: _nodelist_to_slot_render_func(
-                    "<default>",
-                    self.fill_nodes[0].nodelist,
-                    data_var=None,
-                    default_var=None,
-                ),
-            }
-        else:
-            slots = resolve_fill_nodes(context, self.fill_nodes, self.name)
+        slot_fills = resolve_fills(context, self.nodelist, self.name)
```

Comment:
The logic that extracts `{% fill %}` tags from in between the `{% component %} {% endcomponent %}` tags was also reworked, and the logic that assigns the default fill is now inside `resolve_fills()`